src/auth/googleauth.ts

```diff
@@ -682,27 +708,14 @@ export class GoogleAuth {
    * Automatically obtain a client based on the provided configuration.  If no
    * options were passed, use Application Default Credentials.
    */
-  async getClient(options?: GoogleAuthOptions) {
```

Comment:
I don't have a strong opinion about this, but I'm a little bit afraid that this kind of breaking change is breaking in a bad way: it won't actually make the old code (the code that uses `auth.getClient(opts)`) report any errors, but its behavior will be changed.

Do we want to deprecate the options differently by doing this:

```ts
interface DeprecatedOptions {}

async getClient(options?: DeprecatedOptions) {
  if (options) {
    throw new Error('Passing options to getClient is forbidden in v5.0.0. Use new GoogleAuth(opts) instead.');
  }
  ...
}
```

That way we'll deliberately break the code for users that rely on passing options to `getClient` and they will fix the code during the upgrade and not after some long debugging later.

The `DeprecatedOptions` can then be removed completely in v6.0.0.